sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -60,12 +60,28 @@ static const bool UseCopyEngineForD2DCopy = [] {
   return (CopyEngineForD2DCopy && (std::stoi(CopyEngineForD2DCopy) != 0));
 }();
 
+// TODO: Add support for non-zero values of SYCL_PI_LEVEL_ZERO_USE_COPY_ENGINE
 static const bool CopyEngineRequested = [] {
   const char *CopyEngine = std::getenv("SYCL_PI_LEVEL_ZERO_USE_COPY_ENGINE");
-  bool UseCopyEngine = (!CopyEngine || (std::stoi(CopyEngine) != 0));
+  bool UseCopyEngine = (!CopyEngine || ((CopyEngine[1] == '\0') &&
+                                        (std::stoi(CopyEngine) != 0)));
   return UseCopyEngine;
 }();
 
+static const std::pair<int, int> getRangeOfAllowedCopyEngines = [] {
+  const char *CopyEngineRange =
+      std::getenv("SYCL_PI_LEVEL_ZERO_USE_COPY_ENGINE");
+  if (CopyEngineRange && (CopyEngineRange[1] == '\0')) {
+    if (CopyEngineRange[0] == '0')
+      return std::pair<int, int>(-1, -1);
+    else
+      return std::pair<int, int>(0, 8);
```

Comment:
Can we avoid hardcoding 8 here, e.g. have -1 mean no upper bound

Reply:
I replaced this with INT_MAX. That's the right value to use as we want to use ALL copy engines here.